dali/python/nvidia/dali/data_node.py

```diff
@@ -122,6 +139,63 @@ def __bool__(self):
                 " represent computations in DALI Pipeline see the \"Mathematical Expressions\""
                 " section of DALI documentation."))
 
+    def __getitem__(self, val):
```

Comment:
I think that with `__getitem__` defined DataNode is now iterable, you should be able to write:

```
for i in DataNode():
    print("I have no idea what is happening anymore")  
```
and it will do some looping. Unless what we did to prohibit `if DataNode():` still applies here.

Reply:
Fixed - `__iter__` now throws a TypeError.